### DISCUSSION TITLE
The results of FEP are inconsistent with the results of the tutorial

### DISCUSSION THREAD

**Discussion by Zinc-X** [luancarvalhomartins/PyAutoFEP#49] [General]
Dear Prof.

We finished the FEP calculations of FXR (12 to 74, 76, 84, 85, 88) without REST2, but got different results from you(tutorial01/tutorial/ddg_to_center.csv).  We also noticed that the calculated results of 5 ligands are different from the experimental values in the article.
Our calculation results are shown as follows：

             | Our  ddg | Tutorial  |  Exp. Dg
FXR_12 | 0              |             0 | -9.93271
FXR_74 | 0.070849 | 0.109407 | -8.48765
FXR_76 | 0.185977 | -0.23893 | -6.00989
FXR_84 | 0.009766 | -0.16105 | -7.33357
FXR_85 | 0.079347 | 0.511437 | -8.95884
FXR_88 | 0.272636 | 0.759893 | -8.60273
![image](https://user-images.githubusercontent.com/40811569/170233486-8cddcbe9-8fdb-4a23-9f4a-2c544b5ee057.png)

For FEP calculations, all parameters and procedures we use are based on those given in tutorial [OPLS/AA-m, Ligpargen, gromacs, plumed, without REST2/hrex]
Our .ini configuration is shown below

```

# This is the configuration section for prepare_dual_topology.py (which is the only section in this file, but the [ section ] is mandatory)
[prepare_dual_topology]

# Read ligands and topologies from this folder
input_ligands = lig_data

# Read the macromolecule structure from this file
structure = receptor_data/5q17_processed.pdb

# This is the force field directory, it will be copied to each perturbation dir and used to prepare the MD systems
extradirs = oplsaam.ff

# Options controlling the core-constrained superimposition
# First select the use of it instead of reading all ligand poses
pose_loader = superimpose

#lambda set up
lambda_input = lambdas12

# Use this pose as the reference for the superimposition
poses_reference_pose_superimpose = receptor_data/9mv.pdb

# Name of the output. This will be a self-extracting bash file
perturbations_dir = tutorial-t1

# Sets the path to GROMACS executable in the run node, uncomment and modify if needed.
# gmx_bin_run = /usr/local/bin/gromacs

output_hidden_temp_dir=False

# Options controlling the output, see manual for more info. Uncomment as needed
# FEP legs are to be submitted to a slurm scheduler
#output_scripttype = slurm
# Run these commands at the beginnig of the jog (useful to load modules, importing libs)
# output_runbefore = module load python3; module load cuda
# Fine-tune job resources
#output_resources = all_cpus:48; all_gpus:4; all_time: 24
# Use a python file instead of a binary during the collect step
# output_collecttype = python
```

So we would like to ask you if you can give some guidance on how to calculate the simulation accurately. And qe would be deeply grateful if you could give more detailed configuration files of this case.

**Comment by luancarvalhomartins** [comment]
First, you are comparing experimental (absolute) ΔG to calculated (relative) ΔΔG. I redid your table to compare relative to relative free energies (below). Second, due to stochastic nature of MD, each run will sample a different part of the phase space, therefore, yielding a (hopefully) slightly different ΔΔG. So, comparing different runs must be done in this context. Third, comparing to experimental results is even more complex, but your FEP results pretty match ours, which suggests the method to the be consistent.

Lig | Our ΔΔG | Tutorial | Exp. ΔG | Exp. ΔΔG
-- | -- | -- | -- | --
FXR_12 | 0.0 | 0.0 | -9.9 | 0.0
FXR_74 | 0.1 | 0.1 | -8.5 | 1.4
FXR_76 | 0.2 | -0.2 | -6.0 | 3.9
FXR_84 | 0.0 | -0.2 | -7.3 | 2.6
FXR_85 | 0.1 | 0.5 | -9.0 | 1.0
FXR_88 | 0.3 | 0.8 | -8.6 | 1.3

**Reply by Zinc-X** [reply]
Thanks for your reply! We tried this 5 ligand pairs and you mentioned that our results are consistent with yours, but I mean the distinction between them is bad (fluctuating around 0) and they are still quite different  from the calculated values of the three force fields and 14 ligands in the paper. What caused the difference? Where do we need to modify or use non default parameters to fit the gap?  We are willing to try more ligands, such as the 14 FXR ligands but we also think that there are some problems with the FEP parameters we currently use. （I tried more and the results are shown below; I showed the step.int file last time).  

![image](https://user-images.githubusercontent.com/40811569/172278000-3a081041-671c-48ac-b8e4-2ee012b8cb12.png)

**Reply by luancarvalhomartins** [reply]
Ok, first, you results are quite consistent with each other and with the reported results from the tutorial. This is good news and suggests that, as we observed in the paper, PyAutoFEP results are indeed reproducible.

The most likely reason for the difference to the results in paper is that the perturbation map is different (see Fig 4 in the paper vs the figure in the Part 3 of the tutorial). For instance, the ddG prediction for FXR_85 in the paper was obtained by FXR_85->FXR_76->FXR_10 and that of FXR_12 by FXR_12->FXR_76->FXR_10. Comparing these paths to the direct perturbation FXR_12->FXR_85 meay not yield the same results. The ligand set and the perturbations map in the tutorial were selected mostly for simplicity, not because the FEP results are expected to be particularly similar to the experimental data.

Although redoing the calculation with the complete data set (and the same perturbation map) should similar results, I would rather suggest you to retrospectively test in the system you plan to work with. Unless, of course, you are planning to work with the FXR itself. Performance of FEP is highly target-dependent and series-dependent. Furthermore, the FF and the perturbation map affects the result. Even more, what a "good" prediction means also depends on the affinity of the series. Testing on something more similar to the target you will be working with is likely a better idea.

**Reply by Zinc-X** [reply]
I appreciate your suggestion very much. And I am trying using the optimal method for another series which are usually used as benchmarks for FEP caculation. I notice that the perturbation map in Figure 4 is very special compared with the star map, and has many different starting points, not just FXR_10. Also for the optimal method, there are many parameters ,like: optimal_*. So when I choose the "--map_type=optimal",  is it recommended that all other settings remain the default?

**Reply by luancarvalhomartins** [reply]
Thanks for your message. I hope to be able to prepare a tutorial for `generate_perturbation_map.py`, but at the same time, I have been working in a GUI which seems to be a better way to work with perturbation maps anyway.

Most of the default values are fine. Two options are worth exploring: in case you are not interested in cycle closure errors, you can set `optimal_min_edges_per_node` to 1; setting `optimal_max_path` to some value (eg, 4, 5) may render the map more connected.

**Reply by Zinc-X** [reply]
Sorry to trouble you so many times.  I still have 2 questions:

1. Optimal map
I still want to reproduce the results in the article first, so I obtained a total of 14 smiles on D3R web, and then modified the protonation state to -1. However, the resulting smiles and the subsequent generated mol file encountered  errors in the optimal map generation stage` RDkit  AttributeError: 'NoneType' object has no attribute 'GetNumHeavyAtoms'`. I speculate that there may be some mistakes with the smiles. Can you give me a complete input file (include: .smi, step.init) for generating 14 ligands so that I can completely reproduce the calculation tasks of the FXR system

2.  REST2
When I use the REST2 sampling , I can't find any details about the REST2 process ,like temperatures and hamiltonian scaling factors of different replicas. Despite the explanation

>  "where rest2 was used, a scaling factor of 0.5 (26) on the central window was applied to all atoms of the ligand, with a linear scaling towards the end states, and Hamiltonian replica exchange was attempted every 1.0 PS."

What is the scaling factor applied to each lambda window? And how can I set it?
And when I check the md/lambdaX/lambda.log  file, I got this . So dose it mean  they all run at the same temperature?  
![image](https://user-images.githubusercontent.com/40811569/173010594-417c2751-d6c5-4cf5-a8d0-0a89e8cbf820.png)

**Reply by luancarvalhomartins** [reply]
No trouble at all. Feel free to write me or post here whenever you fell like doing so.

1. Sure. I'm sending you later on today. I downloaded the SMILES from the D3R website and generated mol2 files from it, which I used as inputs to `generate_perturbation_map.py`. Just to be sure: `generate_perturbation_map.py` is non-deterministic, so that each run may yield different perturbation maps. In case you would like to run the same perturbation map, you'd rather pass the perturbation map directly to `prepare_perturbation_map.py`.
2. For setting these specific REST2 parameters, please see `hrex` and `solute_scaling`. rest2 script do not use a fictitious temperature for the "hot" region, but rather the beta parameter. See https://github.com/plumed/plumed2/blob/master/scripts/partial_tempering.sh and https://www.tandfonline.com/doi/full/10.1080/00268976.2013.824126. The pressure coupling algorithm uses the same temperature on all replicas, see https://onlinelibrary.wiley.com/doi/full/10.1002/jcc.21703 for original formulation.

**Reply by Zinc-X** [reply]
Thansks！ You can send me ( [zeng_xincheng@163.com](mailto:zeng_xincheng@163.com))

**Reply by Zinc-X** [reply]
Hello, apart from FXR, have you tried any other proteins, such as some classic verification targets
> https://pubs.acs.org/doi/10.1021/ja512751q
, or compare with other FEP calculation methods

**Reply by Zinc-X** [reply]
Hello, I still haven't received the data of FXR target

**Reply by luancarvalhomartins** [reply]
Sorry, I am super swamped these days and didn't find time to get the data from the lab cold storage. I will try to do that tomorrow.

**Reply by luancarvalhomartins** [reply]
> Hello, apart from FXR, have you tried any other proteins, such as some classic verification targets
> 
> > https://pubs.acs.org/doi/10.1021/ja512751q
> > , or compare with other FEP calculation methods

Although retrospective FEP on reported datasets is quite useful, it is my top priority right now, unfortunately. We are, currently, preparing for using PyAutoFEP for prospective studies of targets of medchem interest (namely, MPro and a kinase). Furthermore, I plan to publish the validation of charged perturbations using another series and target. I hope to be able to run other targets at some point, but honestly, it is not something that I think I will be able to work on in the next weeks.
